### PR TITLE
ENT-2803: Adding the use of traverse_pagination in get_content_metadata within the enterprise_catalog API Client

### PR DESCRIPTION
**Pull request by chavesj**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
- [ ] Version pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)

### CODE REVIEW

**Comment by chavesj** [comment]
This change will need to be tested on stage due to the lack of a working configuration for integrated channels in devstack. To test I will be running the stage-edx-transmit_content_metadata Jenkins job with the 'enterprise_worker' user as the argument. This job triggers the transmit_content_metadata management command which uses the get_content_metadata function in which my changes were made.

**Comment by chavesj** [comment]
[Jira Ticket](https://openedx.atlassian.net/browse/ENT-2803)

File: enterprise/api_client/enterprise_catalog.py [hunk 1]
```diff
@@ -128,6 +128,13 @@ def get_content_metadata(self, enterprise_customer, enterprise_catalogs=None):
             endpoint = getattr(self.client, self.GET_CONTENT_METADATA_ENDPOINT.format(catalog_uuid))
             try:
                 response = endpoint.get()
+                results = utils.traverse_pagination(response, endpoint)
```
Comment (bbaker6225) [inline enterprise/api_client/enterprise_catalog.py:131]: You likely already know this, but just a reminder that this won't work until the pagination for `get_content_metadata` in enterprise catalog is actually implemented

File: enterprise/api_client/enterprise_catalog.py [hunk 2]
```diff
@@ -128,6 +128,13 @@ def get_content_metadata(self, enterprise_customer, enterprise_catalogs=None):
...
+                response = {
+                    'count': len(results),
+                    'next': 'None',
+                    'previous': 'None',
+                    'results': results,
+                }
                 for item in response['results']:
```
Comment (iloveagent57) [inline enterprise/api_client/enterprise_catalog.py:138]: Do we actually need `response` here?  Can this be shortened to
```python
response = endpoint.get()
for item in utils.traverse_pagination(response, endpoint):
    ...
```

Comment (chavesj) [reply enterprise/api_client/enterprise_catalog.py:138]: yeah good point, will update

**Pull request merged**